server/src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -284,9 +290,20 @@ public JobDetail demoteContent(
             demotedContent.put(contentId, envContent);
         }
 
-        // We've validated everything here, so we're clear to start deleting...
-        for (EnvironmentContent envContent : demotedContent.values()) {
-            this.envContentCurator.delete(envContent);
+        try {
+            envContentCurator.bulkDelete(new ArrayList<EnvironmentContent>(demotedContent.values()));
```

Comment:
bulkDelete deletes all the content in one transaction. The original code starts and commits a transaction for every delete. Thats a significant difference